api/index.js

```diff
@@ -399,15 +411,73 @@ export const inspectRequest = async (req, res) => {
   })
 }
 
+/**
+ * @param {IncomingMessage} req
+ * @param {ServerResponse} res
+ * @param {pg.Client} client
+ * @param {string} dealIngestionAccessToken
+ */
+export const ingestEligibleDeals = async (req, res, client, dealIngestionAccessToken) => {
+  if (req.headers.authorization !== `Bearer ${dealIngestionAccessToken}`) {
+    res.statusCode = 403
+    res.end('Unauthorized')
+    return
+  }
+
+  const body = await getRawBody(req, { limit: '100mb' })
+  const deals = JSON.parse(body)
+  assert(Array.isArray(deals), 400, 'Invalid JSON Body, must be an array')
+  for (const d of deals) {
+    validate(d, 'clientId', { type: 'string', required: true })
+    validate(d, 'minerId', { type: 'string', required: true })
+    validate(d, 'pieceCid', { type: 'string', required: true })
+    validate(d, 'pieceSize', { type: 'string', required: true })
+    validate(d, 'payloadCid', { type: 'string', required: true })
+    validate(d, 'expiresAt', { type: 'date', required: true })
+  }
+
+  const { rowCount: ingested } = await client.query(`
+    INSERT INTO eligible_deals (
+      client_id,
+      miner_id,
+      piece_cid,
+      piece_size,
+      payload_cid,
+      expires_at,
+      sourced_from_f05_state
+    ) VALUES (
+      unnest($1::TEXT[]),
```

Comment:
There's possibly potential performance improvement here (which we don't have to address now). I have stumbled across this blog post few weeks ago https://www.timescale.com/blog/boosting-postgres-insert-performance

We can do some benchmarks in the future to compare it with current way of how we unnest values.